### PR TITLE
Public Cloud loadBalancer resource

### PR DESCRIPTION
**Pull request by alrayyes**

### CODE REVIEW

**Comment by alrayyes** [comment]
@imtiazPabel I've removed `instance_id` from the loadBalancer resource schema. Unfortunately the image schema can't be reused in loadBalancer schema because there are other differences.

File: internal/provider/publiccloud/load_balancer_resource.go [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("adaptSdkInstanceToResourceInstance: %w", err)
+	}
+	loadBalancer.Contract = contract
+
+	return &loadBalancer, nil
+}
+
+type loadBalancerResource struct {
+	client client.Client
```
Comment (imtiazPabel) [inline internal/provider/publiccloud/load_balancer_resource.go:141]: Would you like to maintain it like `client publicCloud.PublicCloudAPI` for the sake of uniforming?

File: internal/provider/publiccloud/load_balancer_resource.go [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+
+	l.client = coreClient
+}
+
+func (l *loadBalancerResource) Metadata(
+	_ context.Context,
+	request resource.MetadataRequest,
+	response *resource.MetadataResponse,
+) {
+	response.TypeName = request.ProviderTypeName + "_public_cloud_load_balancer"
```
Comment (imtiazPabel) [inline internal/provider/publiccloud/load_balancer_resource.go:183]: Also appreciate, if we can follow the rest of the other resource.

File: internal/provider/publiccloud/load_balancer_resource.go [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, "Launch Public Cloud load balancer")
+
+	opts, err := plan.GetLaunchLoadBalancerOpts(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
```
Comment (imtiazPabel) [inline internal/provider/publiccloud/load_balancer_resource.go:295]: There is a standardize content for summary of the error. I would recommend taking a look at an existing resource.

File: internal/provider/publiccloud/load_balancer_resource.go [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+			httpResponse,
+			err,
+			&response.Diagnostics,
+			context.TODO(),
+		)
+
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Create publiccloud loadBalancer resource for %q", state.ID.ValueString()))
```
Comment (imtiazPabel) [inline internal/provider/publiccloud/load_balancer_resource.go:364]: Typo `Create`

File: internal/provider/publiccloud/load_balancer_resource.go (hunk 2 above)
Comment (alrayyes) [reply internal/provider/publiccloud/load_balancer_resource.go]: This PR is from before we standardized everything

File: internal/provider/publiccloud/load_balancer_resource.go [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+		return
+	}
+
+	tflog.Info(ctx, fmt.Sprintf("Read Public Cloud load balancer %q", state.ID.ValueString()))
+	sdkLoadBalancerDetails, httpResponse, err := l.client.
+		GetLoadBalancer(ctx, state.ID.ValueString()).
+		Execute()
+	if err != nil {
+		utils.HandleSdkError(
+			"Error reading Public Cloud load balancer",
```
Comment (imtiazPabel) [inline internal/provider/publiccloud/load_balancer_resource.go:348]: This summary also needs to be standardized. Please also take into account, if there have any other remaining.

Comment (imtiazPabel) [reply internal/provider/publiccloud/load_balancer_resource.go]: Please take into account, there have more remaining.

**Review by imtiazPabel — APPROVED** [review]

**Pull request merged**